uw-research-computing/testing-and-scaling-up.md

```diff
@@ -0,0 +1,126 @@
+    
+**Requesting Disk Space**
+
+- To inform initial disk requests always look at the size of your input 
+files. At a minimumm, you need to request enough disk to support all 
+of the input files, executable, and the output you expect, but don't forget that the standard 'error' and 'output'
+files you specify will capture 'terminal' output that may add up, too.
+      
+    - If many of your input and output files are compressed 
+(i.e. zipped or tarballs) you will need to factor that into your 
+estimates for disk usage as these files will take up additonal space once uncompressed 
+in the job.
+      
+    - For your initial tests it is OK to request more disk than 
+your job may need so that the test completes successfully. **The key 
+is to adjust disk requests for subsequent jobs based on the results 
+of these test jobs.**
+ 
+**Requesting Memory**
+
+- Estimating **memory requests** can sometimes be tricky. If you've performed the 
+same or similar work on another computer, consider using the amount of 
+memory (i.e. RAM) from that computer as a starting point. For instance, 
+most laptop computers these days will have 8 or 16 GB of memory, which is okay to start 
+with if you know a single job will succeed on your laptop.
+
+	- For your initial tests it is OK to request more memory than 
+your job may need so that the test completes successfully. **The key 
+is to adjust memory requests for subsequent jobs based on the results 
+of these test jobs.**
+
+**Importance of Test Jobs**
+
+- Once you have run a test job using a small number of jobs, **Review the bottom of the HTCondor `log` files from your test jobs to see how many cpus and how much memory and disk space were used.** HTCondor will report 
+the memory, disk, and cpu usage of your jobs in a table at the *bottom* of this file. You can use these values to inform the parameters for future jobs. 
+
+[[[Insert photo of log output here]]]
+
+
+
+# Always Start With Test Jobs
+
+Submitting test jobs is an important first step for optimizing 
+the resource requests of your jobs. We always recommend the following approach whether this is your first time 
+using HTC or you are an experienced user starting a new workflow:
+
+**Step 1: Submit a single test job**
+      - Use a single test job to confirm the job was completed successfully and the results are what you expected.
+
+**Step 2: Submit a few (3-10) test jobs using a single submit file**
+      - Once you have a single test job that completes successfully, the next 
+        step is to submit a small batch of test jobs (e.g. 3 - 10 jobs) 
+        [**using a single submit file**](https://chtc.cs.wisc.edu/uw-research-computing/multiple-jobs). Use this small-scale 
+        multi-job submission test to ensure that all jobs complete successfully, produce the 
+        desired output, and do not conflict with each other when submitted together. 
```

Comment:
Do you want to add a comment here about examining the resources used (as described above in the guide) and adjusting requests if necessary?

Reply:
Done!